tests/sles4sap/redirection_tests/ensa2_kill_sapinstance_ers.pm

```diff
@@ -0,0 +1,109 @@
+
+=cut
+
+sub run {
+    my ($self, $run_args) = @_;
+    my $redirection_data = $run_args->{redirection_data};
+    my $target_hostname = (keys(%{$redirection_data->{nw_ers}}))[0];
+    my %target_data = %{$redirection_data->{'nw_ers'}{$target_hostname}};
+
+    # Connect to target VM
+    connect_target_to_serial(
+        destination_ip => $target_data{ip_address}, ssh_user => $target_data{ssh_user}, switch_root => 1);
+
+    # Get ERS crm resource name
+    my @instance_resources = @{crm_resources_by_class(primitive_class => 'ocf:heartbeat:SAPInstance')};
+    my @ers_resources = grep(/ER/, @instance_resources);
+    die("Exactly one resource must exist. Got:\n" . join("\n", @ers_resources)) unless @ers_resources == 1;
+    my $ers_resource = $ers_resources[0];
+
+    # In some scenarios ERS might be moved to ASCS node - switch to ASCS if that is the case
+    if (crm_resource_locate(crm_resource => $ers_resource) ne $target_hostname) {
+        record_info('NODE SWITCH', "ERS resource '$ers_resource' is on different node, reconnecting.");
+        disconnect_target_from_serial();
+        $target_hostname = (keys(%{$redirection_data->{nw_ascs}}))[0];
+        %target_data = %{$redirection_data->{'nw_ascs'}{$target_hostname}};
+        connect_target_to_serial(
+            destination_ip => $target_data{ip_address}, ssh_user => $target_data{ssh_user}, switch_root => 1);
+    }
+
+    my @instances_data = @{saphostctrl_list_instances(as_root => 'yes')};
+    # Show status
+    sap_show_status_info(cluster => 1, netweaver => 1, instance_id => $instances_data[0]->{instance_id});
+
+    # Check if cluster is being healthy
+    my $fail_count;
+    record_info('Cluster wait', 'Waiting for resources to start');
+    wait_until_resources_started();
+    wait_for_idle_cluster();
+    record_info('Cluster check', 'Checking state of cluster resources');
+    check_cluster_state();
+
+    # Check resource fail count - must be 0
+    $fail_count = crm_get_failcount(crm_resource => $ers_resource, assert_result => 'yes');
+    record_info("Fail count: $fail_count", "Fail count is $fail_count");
+
+    # Kill ENQ process
+    record_info('PROC list', script_output('ps -ef | grep sap'));    # Show SAP processes running
+    my $enq_pid = script_output('pgrep er.sap');
+    die 'ENQ process ID not found' unless $enq_pid;
+    record_info('KILL INST', 'Killing ERS sapinstance process');
+    assert_script_run("kill -9 $enq_pid");
+
+    # Check if ENQ process was killed
+    record_info('PROC list', script_output('ps -ef | grep sap'));    # Show SAP processes running
+
+    script_retry('pgrep er.sap',
+        expect => 1,    # pgrep returns 1 if process was not found
+        delay => 1,    # short delay in case process gets up too quickly
+        timeout => 30,    # 30 sec is plenty
+        fail_message => 'ENQ process still running after being killed.'
+    );
+
+    # Wait for ENQ process to come up again on the same node
```

Comment:
Is this comment correct?
Is it as following?
`s/# Wait for ENQ process to come up again on the same node/# Waiting for cluster detecting failure/`

Reply:
Oh I see... 
The first one was meant in a bigger scope. I'll remove the comment entirely. Looking at it now it's more confusing than necessary. 
Thanks.